pkg/pillar/types/global.go

```diff
@@ -103,6 +110,237 @@ func (gs *GlobalStatus) UpdateItemValuesFromGlobalConfig(gc GlobalConfig) {
 // Agents subscribe to this info to get at least the log levels
 // A value of zero means we should use the default
 // All times are in seconds.
+type ConfigItemType uint8
+
+const (
+	CONFIGITEM_TYPE_INT ConfigItemType = iota + 1
+	CONFIGITEM_TYPE_BOOL
+	CONFIGITEM_TYPE_STRING
+	CONFIGITEM_TYPE_TRISTATE
+)
+
+type ConfigItemSpec struct {
+	Key      string
+	ItemType ConfigItemType
+
+	IntMin     uint32
+	IntMax     uint32
+	IntDefault uint32
+
+	StringDefault   string
+	BoolDefault     bool
+	TriStateDefault TriState
+}
+
+type ConfigItemSpecMap struct {
+	GlobalSettings map[string]ConfigItemSpec
+	AgentSettings  map[string]map[string]ConfigItemSpec
+}
+
+func (configMap *ConfigItemSpecMap) AddIntItem(key string, agentSettingKey string, defaultInt uint32, min uint32, max uint32, agentSetting bool, globalSetting bool) {
+	if defaultInt < min || defaultInt > max {
+		log.Fatalf("Adding int item %s failed. Value does not meet given min/max criteria", key)
+	}
+	configItem := ConfigItemSpec{
+		ItemType:   CONFIGITEM_TYPE_INT,
+		Key:        key,
+		IntDefault: defaultInt,
+		IntMin:     min,
+		IntMax:     max,
+	}
+	if agentSetting {
+		configMap.AgentSettings[key][agentSettingKey] = configItem
+	}
+	if globalSetting {
+		configMap.GlobalSettings[key] = configItem
+	}
+	log.Debugf("Added int item %s", key)
+}
+
+func (configMap *ConfigItemSpecMap) AddBoolItem(key string, agentSettingKey string, defaultBool bool, agentSetting bool, globalSetting bool) {
```

Comment:
This is an unvieldy set of arguments.
AddBoolItem(key string, agent string, defaultBool bool) 
is sufficient. If agent == "" means setting the global, otherwise the agent one.

Ditto for the AddIntItem.